### DISCUSSION TITLE
Add `speed` sensor class

### DISCUSSION THREAD

**Discussion by emontnemery** [home-assistant/architecture#798] [Entity Models] [answered]
## Proposal

Add a `sensor` `device_class` for speed to home assistant core and frontend (+ documentation).

To solve the problem with different applications using different units for speed, without the need to add device classes for each of those applications, the unit of a `speed` sensor shall be configurable by the user in the same way as for temperature and pressure sensors.

Specific device classes for common applications such as wind speed (#775) may of course be added if there is a need.

**Comment by MartinHjelmare** [comment, marked as answer]
Sound good!

**Reply by epenet** [reply]
I opened https://github.com/home-assistant/core/pull/77953
